### PR TITLE
a simple example of using http with graphql-java

### PR DESCRIPTION
**Pull request by bbakerman**
This is a really simple example of using http and graphql-java together.

Its not meant to be production ready but more to point people at when they want an example of http code.

We would then say ... and you can use a myriad of other graphql oriented libraries for production ready code

### CODE REVIEW

**Comment by bbakerman** [comment]
Having this code allow me to better examine our introspection code and make sure it works well with tools such as graphiql.

So it's also handy for testing a bit more end to end.

File: src/test/groovy/example/http/HttpKit.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * in Http Servlet spec world means parsing the query string on POST
+     *
+     * @param request       the http servlet request
+     * @param parameterName the name of the parameter
+     *
+     * @return an optional parameter
+     */
+    public static Optional<String> getQueryParameter(HttpServletRequest request, String parameterName) {
+        if ("POST".equalsIgnoreCase(request.getMethod())) {
+            parameterName = parameterName + "=";
```
Comment (andimarek) [inline src/test/groovy/example/http/HttpKit.java]: why don't we just parse it as json?

Comment (bbakerman) [inline src/test/groovy/example/http/HttpKit.java]: we do but there is an edge case according to the spec.

if its POST BUT there is a query string parameter called query, you should take that instead

Its kinda confusing.  Perhaps we could drop this slavish following of the spec for example purposes

File: src/test/groovy/example/http/HttpMain.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        context.put("YouAppSecurityClearanceLevel", "CodeRed");
+        context.put("YouAppExecutingUser", "Dr Nefarious");
+
+        //
+        // you need a schema in order to execute queries
+        GraphQLSchema schema = buildStarWarsSchema();
+
+        // finally you build a runtime graphql object and execute the query
+        GraphQL graphQL = GraphQL.newGraphQL(schema).build();
+        ExecutionResult executionResult = graphQL.execute(
```
Comment (andimarek) [inline src/test/groovy/example/http/HttpMain.java]: This should use the new `ExecutionInput` object (just merged it).
